vncviewer/ServerDialog.cxx

```diff
@@ -46,74 +46,77 @@
 #include "vncviewer.h"
 #include "parameters.h"
 
+#include "../media/tigervnc.xpm"
 
 using namespace std;
 using namespace rfb;
 
+static Fl_Pixmap tigervnc_icon(tigervnc);
+
 static LogWriter vlog("ServerDialog");
 
 const char* SERVER_HISTORY="tigervnc.history";
 
 ServerDialog::ServerDialog()
-  : Fl_Window(450, 160, _("VNC Viewer: Connection Details"))
+  : Fl_Window(300, 0, "TigerVNC"),
+    usedDir(NULL)
 {
-  int x, y;
+  int y;
+  Fl_Box *box;
   Fl_Button *button;
   Fl_Box *divider;
-
-  int margin = 20;
-  int server_label_width = gui_str_len(_("VNC server:"));
-
-  x = margin + server_label_width;
-  y = margin;
-  
-  serverName = new Fl_Input_Choice(x, y, w() - margin*2 - server_label_width, INPUT_HEIGHT, _("VNC server:"));
-  usedDir = NULL;
-
-  int adjust = (w() - 20) / 4;
-  int button_width = adjust - margin/2;
-
-  x = margin;
-  y = margin + margin/2 + INPUT_HEIGHT;
-
-  y += margin/2;
-
-  button = new Fl_Button(x, y, button_width, BUTTON_HEIGHT, _("Options..."));
-  button->callback(this->handleOptions, this);
-  
-  x += adjust;
-  
-  button = new Fl_Button(x, y, button_width, BUTTON_HEIGHT, _("Load..."));
-  button->callback(this->handleLoad, this);
-
-  x += adjust;
-
-  button = new Fl_Button(x, y, button_width, BUTTON_HEIGHT, _("Save As..."));
-  button->callback(this->handleSaveAs, this);
-  
-  x = 0;
-  y += margin/2 + BUTTON_HEIGHT;
-
-  divider = new Fl_Box(x, y, w(), 2);
+  Fl_Menu_Button *menu;
+
+  y = 20;
+
+  box = new Fl_Box(0, y, w(), tigervnc_icon.h());
+  box->align(FL_ALIGN_CENTER|FL_ALIGN_INSIDE|FL_ALIGN_IMAGE_NEXT_TO_TEXT);
+  box->box(FL_NO_BOX);
+  box->image(tigervnc_icon);
+  box->callback(handleAbout, this);
+  y += box->h() + 20;
+
+  box = new Fl_Box(0, y, w(), 20);
+  box->align(FL_ALIGN_CENTER|FL_ALIGN_INSIDE);
+  box->box(FL_NO_BOX);
+  box->label(_("TigerVNC Viewer"));
+  box->labelsize(20);
+  box->labelfont(FL_BOLD);
+  y += box->h() + 6;
+
+  box = new Fl_Box(0, y, w(), 10);
+  box->align(FL_ALIGN_CENTER|FL_ALIGN_INSIDE);
+  box->box(FL_NO_BOX);
+  box->label(PACKAGE_VERSION);
+  box->labelsize(10);
+  y += box->h() + 40;
+
+  y += INPUT_LABEL_OFFSET;
+  serverName = new Fl_Input_Choice(20, y, w() - 20*2,
+                                   INPUT_HEIGHT, _("Server name:"));
+  serverName->align(FL_ALIGN_LEFT | FL_ALIGN_TOP);
+  y += INPUT_HEIGHT + 20;
+
+  divider = new Fl_Box(20, y, w() - 20*2, 2);
```

Comment:
![tigervnc_no_divider](https://user-images.githubusercontent.com/4117162/151557009-f2937885-7f97-44db-841f-8799ff2822cc.png)

how it looks without a divider

Reply:
Yeah, it seems for example `my_fl_group->color(FL_BLUE);` does nothing.. Setting a background color works on Fl_Box or Fl_Scroll though.

Reply:
This new UI is much better.
Is it possible to add a recent connection list in the main window so that connection can be started by just one double-click instead select from the drop-down list and click OK?
And an option to delete that history in the preferences?